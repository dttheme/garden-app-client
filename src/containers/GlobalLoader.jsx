import React from 'react';
import { connect } from 'react-redux';
import hasToken from '../helpers/token'
import { fetchUserInfo } from "../actions/index.actions";

class GlobalLoader extends React.Component {
  componentDidMount() {
    if (hasToken() && !this.props.user.isLoggedIn) {
      console.log('fetching user info');
      console.log(this.props.user);
      this.props.fetchUserInfo();
    }
}
  render() {
    return (
      <div>
        {this.props.appState.isFetchingUserInfo ?
        <div>
        <h2>
          Loading...
        </h2>
      </div>
      :
      <div>
            {this.props.children}
      </div>
    }
      </div>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user,
  appState: {
    isFetchingUserInfo: state.appState.isFetchingUserInfo,
  },
});

export default connect(mapStateToProps, { fetchUserInfo })(GlobalLoader);
