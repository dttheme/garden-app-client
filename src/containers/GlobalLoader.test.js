import React from 'react';
import { shallow } from 'enzyme';
import { GlobalLoader } from './GlobalLoader';

describe('<GlobalLoader />', () => {

  it('should render page', () => {
    const fakeUser = {
      id: '123',
      username: 'j@e.com',
      firstName: 'Joe',
      isLoggedIn: true,
    }
    const fakeAppState = {
      isFetchingUserInfo: true,
    }
    const wrapper = shallow(<GlobalLoader user={fakeUser} appState={fakeAppState} fetchUserInfo={() => {}} />);
    expect(wrapper).toMatchSnapshot()
  });
});
