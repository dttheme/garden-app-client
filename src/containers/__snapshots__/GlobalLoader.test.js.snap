// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalLoader /> should render page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GlobalLoader
    appState={
        Object {
            "isFetchingUserInfo": true,
          }
    }
    fetchUserInfo={[Function]}
    user={
        Object {
            "firstName": "Joe",
            "id": "123",
            "isLoggedIn": true,
            "username": "j@e.com",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="center"
>
        <h2
                className="mt0"
        >
                Loading...
        </h2>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2
          className="mt0"
>
          Loading...
</h2>,
        "className": "center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
          "className": "mt0",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "h2",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="center"
>
          <h2
                    className="mt0"
          >
                    Loading...
          </h2>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="mt0"
>
            Loading...
</h2>,
          "className": "center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "mt0",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "h2",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
